openedx/adg/lms/webinars/tests/test_helpers.py

```diff
@@ -260,7 +260,26 @@ def test_cancel_all_reminders(msg_id, msg_id_field_name, webinar_registration, m
     setattr(webinar_registration, msg_id_field_name, msg_id)
     cancel_all_reminders([webinar_registration])
 
-    mock_task_cancel_reminders.delay.assert_called_once()
+    assert mock_task_cancel_reminders.delay.call_count == 2
+
+
+@pytest.mark.parametrize(
+    'msg_id_field_name, msg_id', [
+        (STARTING_SOON_REMINDER_ID_FIELD_NAME, FAKE_MANDRILL_MSG_ID),
+        (ONE_WEEK_REMINDER_ID_FIELD_NAME, FAKE_MANDRILL_MSG_ID),
+    ]
+)
+@pytest.mark.django_db
+def test_cancel_all_reminders_when_rescheduling(msg_id, msg_id_field_name, webinar_registration, mocker):
+    """
+    Tests `task_cancel_mandrill_emails` is called synchronously to cancel reminder emails.
+    """
+    mock_task_cancel_reminders = mocker.patch('openedx.adg.lms.webinars.helpers.task_cancel_mandrill_emails')
+
+    setattr(webinar_registration, msg_id_field_name, msg_id)
+    cancel_all_reminders([webinar_registration], is_rescheduling=True)
+
+    assert mock_task_cancel_reminders.call_count == 2
```

Comment:
We can merge this test function with `test_cancel_all_reminders` by passing an extra parameter `is_rescheduling`.